[codegen] Update template for generating TestOnlyClusters

erjiaqing opened the pull request:
#### Problem
What is being fixed?  Examples:

Current codegen for test only clusters limits the codegen for complex context.

#### Change overview

- Rename `DelayCommands` to `TestSuite`
- Create seperate partical for test suite commands for more complex template rules.

#### Testing
- Codegen pass, no (meaningful) code change found.

andy31415 commented on examples/chip-tool/templates/templates.json:
```diff
@@ -26,6 +26,10 @@
             "path": "partials/test_cluster.zapt"
+        },
+        {
+            "name": "TestSuiteHelper_WaitForMs",
```
Could you find a prefix that does not contain things like helper/support/core/util in the name?

andy31415 commented on src/app/tests/suites/TestDelayCommands.yaml:
```diff
@@ -20,7 +20,7 @@ config:
 tests:
     - label: "Wait 100ms"
-      cluster: "DelayCommands"
+      cluster: "TestSuite"
```
Why this rename?

andy31415 commented on src/app/zap-templates/common/ClusterTestGeneration.js:
```diff
@@ -26,7 +26,7 @@ const path              = require('path');
 const templateUtil = require(zapPath + 'dist/src-electron/generator/template-util.js')
 
-const { DelayCommands }                 = require('./simulated-clusters/TestDelayCommands.js');
+const { TestSuiteHelperCluster }        = require('./TestSuiteHelperCluster.js');
```
"TestSuiteHelper" is too much of a generic name, it will end up as a holding bag of everything regardless of related or not. It says "helps tests" and will make code not easily readable in the long run since there  is no obvious mental model of 'helper' (evetyhing is a helper)

andy31415 reviewed (changes requested):
Request changes: please refine naming to be more specific than "helpers" (cannot use util either or support or similar things)

erjiaqing commented:
> Request changes: please refine naming to be more specific than "helpers" (cannot use util either or support or similar things)

Renamed to `SimulatedCluster` (and I think it is better than "DelayCommands"), the reasons are
1. SimulatedCluster can indicate that this "command" is a local-only command, i.e. no new interactions will be invoked between server and client.
2. The command name can already explain what it does, e.g. names `WaitForMs` already explains the function of this command is let the test program wait for a few milliseconds, then SimulatedCluster will limit the scope to the test suite (i.e. it does not send a command to the server and wait for the server's response after a few ms). (Applies to second command)
3. Removed word "TestSuite" to avoid confusion with "TestCluster", which is a *real* cluster.

andy31415 reviewed (approved):
Approving, however overall thought is still that this 'SimulatedCluster' will be a god container of a lot of things. At least simulated somehow implies not a real cluster command.

woody-apple commented:
@msandstedt @saurabhst ?

erjiaqing commented:
Close this PR since we don't need this for now, let's discuss this part later.
